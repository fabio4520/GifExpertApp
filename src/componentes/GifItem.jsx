import PropTypes from 'prop-types'

export const GifItem = ({ url, title }) => {
  return (
    <div className="card">
      <figure>
        <img src={url?? ""} alt={title??""} />
      </figure>
      <p>{ title??"" }</p>
    </div>
  )
}

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}