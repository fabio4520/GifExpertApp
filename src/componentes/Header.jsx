
export const Header = () => {
  return (
    <header className="header">
      <h1 className="header__title">GifExpertApp</h1>
      <nav className="navbar">
        <ul className="navbar__list">
          <li className="navbar__item"> <a href="#trending" className="navbar__link">Trending</a> </li>
          <li className="navbar__item"> <a href="#translate" className="navbar__link">Translate</a> </li>
          <li className="navbar__item"> <a href="#search" className="navbar__link">Search</a> </li>
        </ul>
      </nav>
    </header>
  )
}
