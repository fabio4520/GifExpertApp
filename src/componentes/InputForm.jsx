import PropTypes from 'prop-types'
import { useState } from 'react';

export const InputForm = ({ placeholder, onInputSubmit }) => {
  const [target, setTarget] = useState('')

  const handleSubmit = (e) => { 
    e.preventDefault();
    const newCategory = target.trim();
    if (newCategory.length <= 1) return;
    onInputSubmit(newCategory);
  }

  return (
    <form className="form" onSubmit={handleSubmit} aria-label='form'>
      <input
        type="text"
        className="form__input"
        placeholder={placeholder}
        value={target}
        onChange={(e) => setTarget(e.target.value)}
      />
    </form>
  )
}

InputForm.propTypes = {
  placeholder: PropTypes.string.isRequired,
  onInputSubmit: PropTypes.func.isRequired
}