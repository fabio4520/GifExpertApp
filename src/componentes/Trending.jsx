import { GifItem } from "./GifItem"
import { BsFillArrowLeftCircleFill, BsFillArrowRightCircleFill } from "react-icons/bs";
import { useFetchGifsTrending } from "../hooks/useFetchGifsTrending";

export const Trending = () => {
  
  
  const { gifs, loading, handleNext, handlePrev } = useFetchGifsTrending();
  

  return (
    <section id="trending">
      <h2 className="section__title">Trending</h2>
      <div className="trending-container">
        <button onClick={handlePrev}> <BsFillArrowLeftCircleFill className="arrow arrow--left" /> </button>
        {
          loading
            ? <p>Loading...</p>
            : ( <div className="card-grid">
                  {
                    gifs.map(gif => (
                      <GifItem key={gif.id} url={gif.url} title={ gif.title } />
                    ))
                  }
                </div>)
        }
        <button onClick={handleNext}> <BsFillArrowRightCircleFill className="arrow arrow--right" /> </button>
      </div>
    </section>
  )
}
