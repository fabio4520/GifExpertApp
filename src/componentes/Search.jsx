import { GifItem } from "./GifItem"
import { BsFillArrowLeftCircleFill, BsFillArrowRightCircleFill } from "react-icons/bs";
import { InputForm } from "./InputForm";
import { useFetchGifsSearch } from "../hooks/useFetchGifsSearch";

export const Search = () => {
  
  const { onInputSubmit, handleNext, handlePrev, loading, gifs } = useFetchGifsSearch();
  
  return (
    <section id="search">
      <h2 className="section__title">Search</h2>
      <InputForm placeholder="Find your gifs" onInputSubmit={ onInputSubmit } />
      <div className="search-container">
        <button onClick={handlePrev}> <BsFillArrowLeftCircleFill className="arrow arrow--left" /> </button>
          {
            loading
              ? <p>Loading...</p>
              : ( <div className="card-grid">
                    {
                      gifs.map(gif => (
                        <GifItem key={gif.id} url={gif.url} title={ gif.title } />
                      ))
                    }
                  </div>)
          }
          <button onClick={handleNext}> <BsFillArrowRightCircleFill className="arrow arrow--right" /> </button>
      </div>
    </section>
  )
}
