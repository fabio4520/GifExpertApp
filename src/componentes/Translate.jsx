import { InputForm } from "./InputForm"
import { GifItem } from "./GifItem";
import { useFetchGifTranslate } from "../hooks/useFetchGifTranslate";

export const Translate = () => {

  const { loading, gif, onInputSubmit } = useFetchGifTranslate();  

  return (
    <section id="translate">
      <h2 className="section__title">Translate</h2>
      <InputForm placeholder="Converts words and phrases to the perfect GIF." onInputSubmit={ onInputSubmit } />
      <div className="translate-container" >
        <div className="card-grid">
          {
            loading
              ? <p>Loading...</p>
              : (<GifItem key={gif.id} url={gif.url} title={gif.title} />)
          }
        </div>
      </div>
    </section>
  )
}
