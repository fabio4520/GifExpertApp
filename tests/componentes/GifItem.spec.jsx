import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/componentes/GifItem";

describe('Pruebas en el componente <GifItem />', () => {
  const url = 'https://localhost/algo.jpg';
  const title = 'Un titulo';

  test('debe de mostrar la url enviado por props', () => {
    const { container } = render(<GifItem url={ url } title={ title } />);
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar la imagen con el url y alt indicado', () => {
    render(<GifItem url={url} title={title} />);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  })

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GifItem url={url} title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  })

});
