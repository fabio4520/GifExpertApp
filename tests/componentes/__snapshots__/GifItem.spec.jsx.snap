// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem /> debe de mostrar la url enviado por props 1`] = `
<div>
  <div
    class="card"
  >
    <figure>
      <img
        alt="Un titulo"
        src="https://localhost/algo.jpg"
      />
    </figure>
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
